File: LeetCode/Java/numJewelsInStones.java
/*
    771. Jewels and Stones
    https://leetcode.com/problems/jewels-and-stones/
 */

class Solution {
    public int numJewelsInStones(String J, String S) {
        int count = 0;
        for (int i = 0; i < S.length(); i++) {
            String temp = Character.toString(S.charAt(i));
            if (J.contains(temp))
                count++;
        }
        return count;
    }
}

/*
    Runtime: 1 ms, faster than 93.41% of Java online submissions for Jewels and Stones.
    Memory Usage: 35 MB, less than 97.47% of Java online submissions for Jewels and Stones.
 */

File: LeetCode/Swift/numJewelsInStones.swift
/*
    771. Jewels and Stones
    https://leetcode.com/problems/jewels-and-stones/
 */

class Solution {
    func numJewelsInStones(_ J: String, _ S: String) -> Int {
        var res = 0;
        for c in S {
            if (J.contains(c)) {
                res += 1;
            }
        }
        return res;
    }
}

/*
    Runtime: 8 ms, faster than 78.90% of Swift online submissions for Jewels and Stones.
    Memory Usage: 20.6 MB, less than 7.69% of Swift online submissions for Jewels and Stones.
 */
